test/framework/clusterctl/client.go

```diff
@@ -0,0 +1,155 @@
+		input.CoreProvider,
+		strings.Join(input.BootstrapProviders, ", "),
+		strings.Join(input.ControlPlaneProviders, ", "),
+		strings.Join(input.InfrastructureProviders, ", "),
+	))
+
+	initOpt := clusterctlclient.InitOptions{
+		Kubeconfig:              input.KubeconfigPath,
+		CoreProvider:            input.CoreProvider,
+		BootstrapProviders:      input.BootstrapProviders,
+		ControlPlaneProviders:   input.ControlPlaneProviders,
+		InfrastructureProviders: input.InfrastructureProviders,
+		LogUsageInstructions:    true,
+	}
+
+	clusterctlClient, log := getClusterctlClientWithLogger(input.ClusterctlConfigPath, "clusterctl-init.log", input.LogPath)
+	defer log.Close()
+
+	_, err := clusterctlClient.Init(initOpt)
+	Expect(err).ToNot(HaveOccurred(), "failed to run clusterctl init")
+}
+
+// ConfigClusterInput is the input for ConfigCluster.
+type ConfigClusterInput struct {
+	LogPath                  string
+	ClusterctlConfigPath     string
+	KubeconfigPath           string
+	InfrastructureProvider   string
+	ClusterName              string
+	KubernetesVersion        string
+	ControlPlaneMachineCount *int64
+	WorkerMachineCount       *int64
+}
+
+// ConfigCluster gets a workload cluster based on a template.
+func ConfigCluster(ctx context.Context, input ConfigClusterInput) []byte {
+	By(fmt.Sprintf("clusterctl config cluster %s --infrastructure %s --kubernetes-version %s --control-plane-machine-count %d --worker-machine-count %s",
+		input.ClusterName,
+		input.InfrastructureProvider,
+		input.KubernetesVersion,
+		input.ControlPlaneMachineCount,
+		input.WorkerMachineCount,
+	))
+
+	templateOptions := clusterctlclient.GetClusterTemplateOptions{
+		Kubeconfig: input.KubeconfigPath,
+		ProviderRepositorySource: &clusterctlclient.ProviderRepositorySourceOptions{
+			InfrastructureProvider: input.InfrastructureProvider,
+			Flavor:                 "",
+		},
+		ClusterName:              input.ClusterName,
+		KubernetesVersion:        input.KubernetesVersion,
+		ControlPlaneMachineCount: input.ControlPlaneMachineCount,
+		WorkerMachineCount:       input.WorkerMachineCount,
+	}
+
+	clusterctlClient, log := getClusterctlClientWithLogger(input.ClusterctlConfigPath, "clusterctl-config-cluster.log", input.LogPath)
+	defer log.Close()
+
+	template, err := clusterctlClient.GetClusterTemplate(templateOptions)
+	Expect(err).ToNot(HaveOccurred(), "Failed to run clusterctl config cluster")
+
+	yaml, err := template.Yaml()
+	Expect(err).ToNot(HaveOccurred(), "Failed to generate yaml for the workload cluster template")
+
+	log.WriteString(string(yaml))
+
+	return yaml
+}
+
+// MoveInput is the input for ClusterctlMove.
+type MoveInput struct {
+	LogPath              string
+	ClusterctlConfigPath string
+	FromKubeconfigPath   string
+	ToKubeconfigPath     string
+	Namespace            string
+}
+
+// ClusterctlMove moves a workloads clusters.
+func ClusterctlMove(ctx context.Context, input MoveInput) {
```

Comment:
nit. Rename to Move to be consistent with Init()?

Reply:
done!